packages/neo-one-smart-contract-compiler/src/compile/syscalls.ts

```diff
@@ -977,6 +978,15 @@ export const SYSCALLS = {
     name: 'Neo.Storage.GetContext',
     returnType: StorageContextValue,
   }),
+  'Neo.Storage.GetReadOnlyContext': new SimpleSysCall({
+    name: 'Neo.Storage.GetReadOnlyContext',
+    returnType: StorageContextReadonlyValue,
+  }),
+  'Neo.StorageContext.AsReadOnly': new SimpleSysCall({
+    name: 'Neo.StorageContext.AsReadOnly',
+    args: [new SysCallArgument('context', new UnionValue([StorageContextValue, StorageContextReadonlyValue]))],
+    returnType: StorageContextReadonlyValue,
+  }),
   'Neo.Storage.Get': new SimpleSysCall({
     name: 'Neo.Storage.Get',
     args: [new SysCallArgument('context', StorageContextValue), new SysCallArgument('key', StorageKey)],
```

Comment:
This should take a union of `StorageContextValue` and `StorageContextReadonlyValue`